packages/parser/src/helpers/dynamodb.ts

```diff
@@ -0,0 +1,82 @@
+import type { AttributeValue } from '@aws-sdk/client-dynamodb';
+import { unmarshall } from '@aws-sdk/util-dynamodb';
+import { type ZodTypeAny, z } from 'zod';
+
+/**
+ * A helper function to unmarshall DynamoDB stream events and validate them against a schema.
+ *
+ * @example
+ * ```typescript
+ * const mySchema = z.object({
+ *   id: z.string(),
+ *   name: z.string(),
+ * });
+ * const eventSchema = DynamoDBStreamSchema.extend({
+ *   Records: z.array(
+ *     DynamoDBStreamRecord.extend({
+ *       dynamodb: z.object({
+ *         NewImage: DynamoDBMarshalled(mySchema).optional(),
+ *       }),
+ *     })
+ *   ),
+ * });
+ * type eventSchema = z.infer<typeof extendedSchema>;
+ * ```
+ * For example, if you have a DynamoDB stream event like the following:
+ *
+ * ```json
+ * {
+ *   "Records": [
+ *     {
+ *       "dynamodb": {
+ *         "NewImage": {
+ *           "id": {
+ *             "S": "12345"
+ *           },
+ *           "name": {
+ *             "S": "John Doe"
+ *           }
+ *         }
+ *       }
+ *     }
+ *   ]
+ * }
+ * ```
+ * Resulting in:
+ *
+ * ```json
+ * {
+ *   "Records": [
+ *     {
+ *       "dynamodb": {
+ *         "NewImage": {
+ *           "id": "12345",
+ *           "name": "John Doe"
+ *         }
+ *       }
+ *     }
+ *   ]
+ * }
+ * ```
+ *
+ * @param schema - The schema to validate the JSON string against
+ */
+const DynamoDBMarshalled = <T extends ZodTypeAny>(schema: T) =>
+  z
+    .record(z.string(), z.custom<AttributeValue>())
```

Comment:
I would appreciate suggestions on using this type.

Setting the type to `unknown` causes a TypeScript error. The `@aws-sdk/util-dynamodb` package has a peer dependency on `@aws-sdk/client-dynamodb` and we are using the `AttributeValue` type from `@aws-sdk/client-dynamodb`.  Should we also declare `@aws-sdk/client-dynamodb` as a peer dependency? 

Alternatively, we could set the type to `unknown` and suppress the TypeScript error using `biome-ignore`, though I would prefer to avoid this approach.